src/controllers/confirm.company.controller.ts

```diff
@@ -16,11 +21,29 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = async (req: Request, res: Response, next: NextFunction) => {
+export const post = async (req: Request, res: Response) => {
+  const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
```

Comment:
i can't see the benefit of getting the companyProfile as we only seem to be using the companyNumber from it. We already have the company number as we pass that into getCompanyProfile as a param.
You can remove the getCompanyProfile and use the companyNumber from the req